### DISCUSSION TITLE
How to contribute to the project?

### DISCUSSION THREAD

**Discussion by kone9** [nxrighthere/UnrealCLR#173] [General]
I wanted to know how I can do to contribute.
There is a possibility that you will receive an epic mega grant to help with this development.

I am a user of Godot Engine, Unity3D and Unreal Engine.
I find problems in all the engines.

In Godot Engine the engine node layout is great, but the C # support is pretty bad and the 3D graphics engine is not good. The worst thing about Godot Engine is that it lacks the professional tools that Unreal Engine has for 3D development.

In Unity I find a good engine, a design not as simple as the Godot Engine nodes, but quite easy to use. The problem is that the graphics engine does not reach the level of Unreal Engine and it does not have the same tools.

In Unreal Engine I find a great graphics engine and the best tools, but it fails in the other one, "OSEA in USABILITY". Blueprint dynamism is good, but it will never be the same as writing Sharp C code.
The problem is that blueprints are slow to program and writing the code is always faster. I always hear the creators of Unreal Engine say that nodes are for NON-PROGRAMMERS; but actually using the nodes is programming, but visually.
It is a pity that Epic Games has removed the old language that it used in UDK "UnrealScript".

For the Unreal Engine with CPP issue:
CPP is too slow for development with Unreal Engine, intillicence in IDES like VSCode or Visual Studio works extremely poorly, compilation times and concepts behind tools are a real headache when it comes to interpreting all concepts with CPP.

People use a video game engine to "MAKE GAMES", it is not necessary to level down ..
I know there are many programmers who like to boast that they know how to program in CPP, but making a video game is much more than understanding pointers. I don't think it's worth that headache.

The creators of Unreal Engine know this very well, that's why they made the blueprints and used UnrealScript before. For a reason, Unity with C # is practically the standard nowadays for making indie games.

So I think an API in CSharp would solve a lot of problems.

I see you have a lot of work to do. 
![Screenshot_4](https://user-images.githubusercontent.com/12847027/129047346-75be0bc8-013c-4349-b3e1-e141b9b75e4b.jpg)

I want to take this as a job and develop C # Export compatibility to HTML, Android and Desktop ... mostly to guide indie development.
How can I help you and get an epic Megagrant?

# I write in Spanish too because it is my original language.
# Unreal engine con C sharp

Quería saber cómo puedo hacer para contribuir.
Existe la posibilidad de que reciba una mega subvención épica para ayudar con este desarrollo.

Soy usuario de Godot Engine, Unity3D y Unreal Engine.
Encuentro problemas en todos los motores.

En Godot Engine el diseño de nodos del motor es excelente, pero el soporte para C # es bastante malo y el motor de gráficos 3D no es bueno. Lo peor de Godot Engine es que carece de las herramientas profesionales que tiene Unreal Engine para el desarrollo 3D.

En Unity encuentro un buen motor, un diseño no tan simple como los nodos de Godot Engine, pero bastante fácil de utilizar, El problema es que el motor gráfico no llega al nivel de Unreal Engine y tampoco tiene las mismas herramientas.

En Unreal Engine encuentro un gran motor gráfico y las mejores herramientas, pero falla en el otro, "OSEA en la USABILIDAD". El dinamismo de los blueprint es bueno, pero nunca será lo mismo que escribir código C Sharp.
El problema es que los blueprint son lentos para programar y la escritura del código siempre es mas rápida. Escucho siempre decir a los creadores de Unreal Engine, que los nodos son para NO PROGRAMADORES; pero en realidad usar los nodos es programar, pero de forma visual.
Es una pena que Epic Games haya sacado el viejo lenguaje que usaba en UDK "UnrealScript".

Por el tema de Unreal Engine con CPP:
CPP es demasiado lento para el desarrollo con Unreal Engine, el intillicence en los IDES como VSCode o Visual Studio funciona extremadamente mal, los tiempos de compilación y los conceptos detrás de las herramientas son un verdadero dolor de cabeza cuando se trata de interpretar todos los conceptos con CPP.

La gente usa un motor de videojuegos para "HACER JUEGOS", no es necesario bajar de nivel..
Se que hay muchos programadores que les gusta alardear de que saben programar en CPP, pero hacer un videojuego es mucho más que comprender los punteros. No creo que valga la pena ese dolor de cabeza.

Los creadores de Unreal Engine esto lo saben muy bien, por eso hicieron los blueprint y antes usaban UnrealScript. Por algo será que Unity con C# es prácticamente el standart hoy en día para hacer juegos indies.

Por eso creo que una API en CSharp solucionaría muchos problemas.

Veo que tienes mucho trabajo por hacer.
Screenshot_4

Quiero tomar esto como un trabajo y desarrollar compatibilidad con C # Export a HTML, Android y Desktop ... Más que nada para guiar el desarrollo indie. 
¿Cómo puedo hacer para ayudarte y obtener un Megagrante épico?

**Comment by nxrighthere** [comment]
> I wanted to know how I can do to contribute.

The best contribution for this project is feedback, I prioritize tasks based on that.

**Reply by kone9** [reply]
> > I wanted to know how I can do to contribute.
> 
> The best contribution for this project is feedback, I prioritize tasks based on that.

Ok, I am going to test the tool and then I will comment on the problems here. Glad you answer the questions.

**Reply by darkguy2008** [reply]
@kone9 your points are pretty valid and @nxrighthere can count on me too to help him get the Epic Megagrant or as much as help as possible as long as it's within my reach. I agree 10000000% with all you say as those have been the major pain points from other engines and Epic's as well, C++ isn't just worth it and it saddens me to see they keep going with it instead of having support for more user-friendly languages like C# but instead they just keep throwing blueprints in our face as a replacement when it's not, and it's clunky as heck.

I really hope UnrealCLR fills this gap, I'm gonna give it a try, but count me in for anything you need!
